### PR TITLE
ROS plugins use plugins' build directory

### PR DESCRIPTION
**Pull request by artivis**
ROS plugins use the plugins' build directory introduced in #3347 to dump their build artifact while the `part-name/build` directory is only used to fetch source files.

This allows for distributing the `snapcraft.yaml` config file directly alongside a ROS package (in the package root directory) whereas it can currently only be distributed with a ROS workspace.

Follows #3347.

Signed-off-by: artivis deray.jeremie@gmail.com

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/plugins/v2/_ros.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def stage_runtime_dependencies(part_install: str, ros_distro: str):
     # TODO: support python packages (only apt currently supported)
     apt_packages: Set[str] = set()
 
-    for pkg in catkin_packages.find_packages(".").values():
+    for pkg in catkin_packages.find_packages("$SNAPCRAFT_PART_BUILD_WORK").values():
```
Comment (kyrofa) [inline snapcraft/plugins/v2/_ros.py:136]: This isn't bash, you'll need to pass in the directory to the class.

Comment (artivis) [reply snapcraft/plugins/v2/_ros.py:136]: My bad, I actually didn't mean to push this since it is not necessary. I'll revert it.

Comment (artivis) [reply snapcraft/plugins/v2/_ros.py]: Reverted in 12087c9.

**Comment by artivis** [comment]
Closing this as we go for a different scheme. See https://github.com/snapcore/snapcraft/pull/3347.

**Pull request closed without merging**